frontend/tests/controllers-test/challengeListCtrl.test.js

```diff
@@ -0,0 +1,282 @@
+                (isUpcomingChallengeSucess == false && parameters.url == 'challenges/challenge/future') ||
+                (isPastChallengeSuccess == false && parameters.url == 'challenges/challenge/past')){
+                    parameters.callback.onError({
+                        data: errorResponse
+                    });
+                }
+            };
+        });
+
+        it('when no ongoing challenge found `challenges/challenge/present`', function () {
+            isPresentChallengeSuccess = true;
+            isUpcomingChallengeSucess = null;
+            isPastChallengeSuccess = null;
+            successResponse = {
+                results: []
+            };
+            vm = createController();
+            expect(vm.currentList).toEqual(successResponse.results);
+            expect(vm.noneCurrentChallenge).toBeTruthy();
+        });
+
+        it('check description length and calculate timezone of ongoing challenge `challenges/challenge/present`', function () {
+            isPresentChallengeSuccess = true;
+            isUpcomingChallengeSucess = null;
+            isPastChallengeSuccess = null;
+            successResponse = {
+                results: [
+                    {
+                        id: 1,
+                        description: "the length of the ongoing challenge description is greater than or equal to 50",
+                        creator: {
+                            id: 1
+                        },
+                        start_date: "Fri June 12 2018 22:41:51 GMT+0530",
+                        end_date: "Fri June 12 2099 22:41:51 GMT+0530"
+                    },
+                    {
+                        id: 2,
+                        description: "random description",
+                        creator: {
+                            id: 1
+                        },
+                        start_date: "Sat May 26 2015 22:41:51 GMT+0530",
+                        end_date: "Sat May 26 2099 22:41:51 GMT+0530"
+                    }
+                ]
+            };
+            vm = createController();
+            expect(vm.currentList).toEqual(successResponse.results);
+            expect(vm.noneCurrentChallenge).toBeFalsy();
+
+            var timezone = moment.tz.guess();
+            for (var i in vm.currentList) {
+                if (vm.currentList[i].description.length >= 50) {
+                    expect(vm.currentList[i].isLarge).toEqual("...");
+                } else {
+                    expect(vm.currentList[i].isLarge).toEqual("");
+                }
+                var offset = new Date(vm.currentList[i].start_date).getTimezoneOffset();
+                expect(vm.currentList[i].start_zone).toEqual(moment.tz.zone(timezone).abbr(offset));
+                offset = new Date(vm.currentList[i].end_date).getTimezoneOffset();
+                expect(vm.currentList[i].end_zone).toEqual(moment.tz.zone(timezone).abbr(offset));

```

Comment:
extract out `moment.tz.zone(timezone)` to outside of for loop, that way we don't have to re-calculate it every time.

Reply:
@galipremsagar hey, but `offset` is changing according to the `start_date` so we cannot move this outside the loop